### ISSUE TITLE
Resample introduces a time lag

### ISSUE THREAD

**Issue by CommonClimate** [LinkedEarth/Pyleoclim_util#361]
**Describe the bug**
the default behavior of resample() introduces a time lag (at least in conjunction with averaging) which is undesirable. 

**To Reproduce**
See the interactive plot at the bottom of [Resampling_Tutorial.ipynb](https://github.com/LinkedEarth/paleoPandas/tree/main/notebooks)
![](https://user-images.githubusercontent.com/13760223/222862838-ee96c7eb-36b0-452a-93ed-16a25299231c.png)

**Expected behavior**
shift the index by 1/2 an interval, so that resampled peaks/troughs (better) coincide with originals, instead of having a systematic offset

**Comment by MarcoGorelli** [comment]
> shift the index by 1/2 an interval

this could be done, just by inserting
```python
series.index = series.index + (series.index[1] - series.index[0])/2
```
after this line

https://github.com/LinkedEarth/Pyleoclim_util/blob/17ea7a1adb0290c4eaa002e70e0c5bf10d5493bf/pyleoclim/core/series.py#L4383

However, they still wouldn't be perfectly aligned. Say your index starts off as
```python
'-5318046-05-22 11:46:11', '-5313046-05-20 11:37:24'
```
After resampling by 5000 years, the index becomes
```
'-5318046-01-01', '-5313046-01-01'
```
If you then shift by `(series.index[1] - series.index[0]) / 2`, then the first one would become
```python
'-5315547-12-31 12:00:00'
```
which still wouldn't align with the original value of `'-5318046-05-22 11:46:11'`

**Comment by CommonClimate** [comment]
The indices are off by <6months, which is a LOT better than 5000 years ;-)  I say that is worth the inexactitude. Can you implement that in the code, pls? I'm not envisioning a numeric test, just the "eye test" on the series above would be enough.

**Comment by CommonClimate** [comment]
Or I can try to do it myself, hoping I won't break anything...

**Comment by MarcoGorelli** [comment]
sure, up to you, feel free to open a pull request and tag me for review, please let me know what you'd prefer

**Comment by CommonClimate** [comment]
Done. Will make the PR momentarily. It is a thing of beauty, but now all 11 resample unit tests are failing. They were written in a way that hard-coded the original assumption, and I'd appreciate if you can fix them. Thank you.
![LR04_resample5k](https://user-images.githubusercontent.com/13760223/229256154-db25d68f-011e-49fd-b62f-a3d83eef28b5.png)

**Comment by CommonClimate** [comment]
fixed by ba1c79f4183858bab0beca2d57ba551dce16b3bb
